src/devtools/views/Components/TreeContext.js

```diff
@@ -380,15 +379,11 @@ function reduceSearchState(store: Store, state: State, action: Action): State {
           store.roots.forEach(rootID => {
             recursivelySearchTree(store, rootID, regExp, searchResults);
           });
-
           if (searchResults.length > 0) {
-            if (prevSearchIndex === null) {
-              searchIndex = 0;
+            if (selectedElementID !== null) {
+              searchIndex = getNearestResult(searchResults, selectedElementID);
             } else {
-              searchIndex = Math.min(
-                ((prevSearchIndex: any): number),
-                searchResults.length - 1
-              );
```

Comment:
Hm, I _think_ we might want to preserve this behavior since it would be less of a jump in some cases.

1. Go to [the demo page](https://react-devtools-experimental.now.sh/)
2. Search for "ModernContext" and hit enter twice to select the third result
3. Type an additional "." (so you're now searching for "ModernContext.")

Before this change, the selected index would go from 3 to 2 (the nearest result given the updated text):
![search-before-Kapture 2019-07-27 at 8 50 10](https://user-images.githubusercontent.com/29597/61996667-9af08300-b04b-11e9-9b89-699ecddb9212.gif)

After this change, the cursor will jump back to index 0 (the furthest result).
![search-after-Kapture 2019-07-27 at 8 51 35](https://user-images.githubusercontent.com/29597/61996679-cd9a7b80-b04b-11e9-9f3f-04bc1a9ca2b6.gif)

To be fair, this is a minor thing- but I think it's slightly nicer if we preserve the original behavior for this case. I think we can do that with just a minor tweak.

```diff
diff --git a/src/devtools/views/Components/TreeContext.js b/src/devtools/views/Components/TreeContext.js
index 5af9398..17b1a5c 100644
--- a/src/devtools/views/Components/TreeContext.js
+++ b/src/devtools/views/Components/TreeContext.js
@@ -277,6 +277,7 @@ function reduceSearchState(store: Store, state: State, action: Action): State {
     selectedElementIndex,
   } = state;
 
+  const prevSearchIndex = searchIndex;
   const prevSearchText = searchText;
   const numPrevSearchResults = searchResults.length;
 
@@ -380,10 +381,20 @@ function reduceSearchState(store: Store, state: State, action: Action): State {
             recursivelySearchTree(store, rootID, regExp, searchResults);
           });
           if (searchResults.length > 0) {
-            if (selectedElementID !== null) {
-              searchIndex = getNearestResult(searchResults, selectedElementID);
+            if (prevSearchIndex === null) {
+              if (selectedElementID !== null) {
+                searchIndex = getNearestResult(
+                  searchResults,
+                  selectedElementID
+                );
+              } else {
+                searchIndex = 0;
+              }
             } else {
-              searchIndex = 0;
+              searchIndex = Math.min(
+                ((prevSearchIndex: any): number),
+                searchResults.length - 1
+              );
             }
           }
         }
```

Since I'm already in the code I'll just go ahead and make this tweak. If you know of a good reason why we *shouldn't* do this though, let me know?